segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperServiceRunner.java

```diff
@@ -442,29 +487,55 @@ public ResourceBuilder(ServerConfiguration conf) {
             this.conf = conf;
         }
 
-        BookieResources build() throws Exception {
+        BookieResources build() {
             if (metadataBookieDriver == null) {
-                if (conf.getMetadataServiceUri() == null) {
-                    metadataBookieDriver = new NullMetadataBookieDriver();
-                } else {
-                    metadataBookieDriver = org.apache.bookkeeper.bookie.BookieResources.createMetadataDriver(conf, NullStatsLogger.INSTANCE);
+                try {
+                    if (conf.getMetadataServiceUri() == null) {
+                        metadataBookieDriver = new NullMetadataBookieDriver();
+                    } else {
+                        metadataBookieDriver = org.apache.bookkeeper.bookie.BookieResources.createMetadataDriver(conf, NullStatsLogger.INSTANCE);
+                    }
+                } catch (ConfigurationException e) {
+                    throw new RuntimeException(e);
+                } catch (BookieException e) {
+                    throw new RuntimeException(e);
                 }
             }
             if (registrationManager == null) {
                 registrationManager = metadataBookieDriver.createRegistrationManager();
             }
-            LedgerManagerFactory ledgerManagerFactory = metadataBookieDriver.getLedgerManagerFactory();
+            LedgerManagerFactory ledgerManagerFactory = null;
+            try {
+                ledgerManagerFactory = metadataBookieDriver.getLedgerManagerFactory();
+            } catch (MetadataException e) {
+                throw new RuntimeException(e);
+            }
             LedgerManager ledgerManager = ledgerManagerFactory.newLedgerManager();
 
             DiskChecker diskChecker = org.apache.bookkeeper.bookie.BookieResources.createDiskChecker(conf);
-            LedgerDirsManager ledgerDirsManager = org.apache.bookkeeper.bookie.BookieResources.createLedgerDirsManager(
-                    conf, diskChecker, NullStatsLogger.INSTANCE);
-            LedgerDirsManager indexDirsManager = org.apache.bookkeeper.bookie.BookieResources.createIndexDirsManager(
-                    conf, diskChecker, NullStatsLogger.INSTANCE, ledgerDirsManager);
-
-            LedgerStorage storage = org.apache.bookkeeper.bookie.BookieResources.createLedgerStorage(
-                    conf, ledgerManager, ledgerDirsManager, indexDirsManager,
-                    NullStatsLogger.INSTANCE, UnpooledByteBufAllocator.DEFAULT);
+            LedgerDirsManager ledgerDirsManager = null;
```

Comment:
Can you just have a single catch block for all the below? Looks like repeating the same catch block over and over again for consecutive lines.

Reply:
updated with single catch block